packages/main/src/plugin/container-registry.spec.ts

```diff
@@ -1530,6 +1554,80 @@ describe('buildImage', () => {
     });
   });
 
+  test('verify containerfile is added to archive if outside of context', async () => {
+    const dockerAPI = new Dockerode({ protocol: 'http', host: 'localhost' });
+
+    // set providers with docker being first
+    containerRegistry.addInternalProvider('podman1', {
+      name: 'podman',
+      id: 'podman1',
+      api: dockerAPI,
+      libpodApi: dockerAPI,
+      connection: {
+        type: 'podman',
+        endpoint: {
+          socketPath: '/endpoint1.sock',
+        },
+        name: 'podman',
+      },
+    } as unknown as InternalContainerProvider);
+
+    const connection: ProviderContainerConnectionInfo = {
+      name: 'podman',
+      displayName: 'podman',
+      type: 'podman',
+      endpoint: {
+        socketPath: '/endpoint1.sock',
+      },
+      lifecycleMethods: undefined,
+      status: 'started',
+    };
+
+    vi.spyOn(util, 'isWindows').mockImplementation(() => false);
+    vi.spyOn(tar, 'pack').mockImplementation(originalTarPack);
+    vi.spyOn(dockerAPI, 'buildImage').mockResolvedValue({} as NodeJS.ReadableStream);
+    vi.spyOn(dockerAPI.modem, 'followProgress').mockImplementation((_s, f, _p) => {
+      return f(null, []);
+    });
+
+    vi.mocked(fs.existsSync).mockImplementation(path => {
+      return String(path).endsWith('Containerfile.0') || String(path).endsWith('Containerfile.1');
+    });
+    vi.mocked(dockerAPI.buildImage).mockReset();
+
+    contextDir = await mkdtemp(path.join(os.tmpdir(), 'pd-tests-'));
```

Comment:
I don't see why we need to create a directory. why are we not mocking all the fs operations ?
it will avoid to have to delete some folders at the end

Reply:
right, it got me some time to understand that `tar-fs` module uses `lstat.call(...)` to call `lstat`, and because of this the mocks are not working. Fortunately `tar.pack` accepts an optional `fs`, to pass the mocked fs.

I changed to use this optional `fs`